internal/provider/kubernetes/controller.go

```diff
@@ -1543,132 +1548,143 @@ func (r *gatewayAPIReconciler) enqueueClass(_ context.Context, _ client.Object)
-	if ep.Namespace != r.namespace {
-		r.log.Info("envoyproxy namespace does not match Envoy Gateway's namespace",
-			"namespace", ep.Namespace, "name", ep.Name)
-		return false
+// processGatewayParamsRef processes the infrastructure.parametersRef of the provided Gateway.
+func (r *gatewayAPIReconciler) processGatewayParamsRef(ctx context.Context, gtw *gwapiv1.Gateway, resourceMap *resourceMappings, resourceTree *gatewayapi.Resources) error {
+	if gtw == nil || gtw.Spec.Infrastructure == nil || gtw.Spec.Infrastructure.ParametersRef == nil {
+		return nil
 	}
 
-	gcList := new(gwapiv1.GatewayClassList)
-	err := r.client.List(context.TODO(), gcList)
-	if err != nil {
-		r.log.Error(err, "failed to list gatewayclasses")
-		return false
+	if resourceTree.ClassEnvoyProxy != nil && resourceTree.ClassEnvoyProxy.Spec.MergeGateways != nil && *resourceTree.ClassEnvoyProxy.Spec.MergeGateways {
+		return fmt.Errorf("infrastructure.parametersref must be nil when MergeGateways feature is enabled by gatewayclass")
 	}
 
-	for i := range gcList.Items {
-		gc := gcList.Items[i]
-		// Reconcile the managed GatewayClass if it's referenced by the EnvoyProxy.
-		if r.hasMatchingController(&gc) &&
-			classAccepted(&gc) &&
-			classRefsEnvoyProxy(&gc, ep) {
-			return true
+	ref := gtw.Spec.Infrastructure.ParametersRef
+	if !(string(ref.Group) == egv1a1.GroupVersion.Group &&
+		ref.Kind == egv1a1.KindEnvoyProxy &&
+		len(ref.Name) > 0) {
+		return fmt.Errorf("unsupported parametersRef for gateway %s/%s", gtw.Namespace, gtw.Name)
+	}
+
+	ep := new(egv1a1.EnvoyProxy)
+	if err := r.client.Get(ctx, types.NamespacedName{Namespace: gtw.Namespace, Name: ref.Name}, ep); err != nil {
+		return fmt.Errorf("failed to find envoyproxy %s/%s: %w", gtw.Namespace, ref.Name, err)
+	}
+
+	if err := r.processEnvoyProxy(ep, resourceMap); err != nil {
+		return err
+	}
+
+	if ep.Spec.BackendTLS != nil && ep.Spec.BackendTLS.ClientCertificateRef != nil {
+		certRef := ep.Spec.BackendTLS.ClientCertificateRef
+		if refsSecret(certRef) {
+			if err := r.processSecretRef(
+				ctx,
+				resourceMap,
+				resourceTree,
+				gatewayapi.KindGateway,
+				gtw.Namespace,
+				gtw.Name,
+				*certRef); err != nil {
+				r.log.Error(err,
+					"failed to process TLS SecretRef for gateway",
+					"gateway", utils.NamespacedName(gtw).String(), "secretRef", certRef)
+			}
 		}
 	}
 
-	return false
+	resourceTree.EnvoyProxies = append(resourceTree.EnvoyProxies, ep)
+	return nil
 }
 
-// processParamsRef processes the parametersRef of the provided GatewayClass.
-func (r *gatewayAPIReconciler) processParamsRef(ctx context.Context, gc *gwapiv1.GatewayClass, resourceMap *resourceMappings, resourceTree *gatewayapi.Resources) error {
+// processGCParamsRef processes the parametersRef of the provided GatewayClass.
+func (r *gatewayAPIReconciler) processGCParamsRef(ctx context.Context, gc *gwapiv1.GatewayClass, resourceMap *resourceMappings, resourceTree *gatewayapi.Resources) error {
```

Comment:
rethink about it, in order to keep clear, let's better not to use the abbreviation of GatewayClass 

```suggestion
func (r *gatewayAPIReconciler) processGatewayClassParamsRef(ctx context.Context, gc *gwapiv1.GatewayClass, resourceMap *resourceMappings, resourceTree *gatewayapi.Resources) error {
```